src/pages/vote/create.tsx

```diff
@@ -1,312 +1,279 @@
   InputRightElement,
-  InputLeftElement 
-} from "@chakra-ui/react";
-import snapshot from '@snapshot-labs/snapshot.js';
-import {FaWindowClose} from 'react-icons/fa';
-import { uuid } from 'uuidv4';
-import { Web3Provider } from '@ethersproject/providers';
-import { useAuth } from './../../services/context/auth';
-import { useForm } from "react-hook-form";
+} from '@chakra-ui/react'
+import snapshot from '@snapshot-labs/snapshot.js'
+import { FaWindowClose } from 'react-icons/fa'
+import { Web3Provider } from '@ethersproject/providers'
 
 const hub = 'https://hub.snapshot.org'
-const client = new snapshot.Client712(hub);
+const client = new snapshot.Client712(hub)
 
 export default function Create() {
-  const [account, setAccount] = React.useState<any>("")
-  const web3 = new Web3Provider(window.ethereum);
-  const [ snapshotClient ] = React.useState<any>(client);
-  const [ title, setTitle ] = React.useState<string>("");
-  const [ content, setContent ] = React.useState<string>("");
-  const [ snapshotDate, setSnapShotDate ] = React.useState<any>(new Date());
-  const [ startDate, setStartDate ] = React.useState<any>(new Date());
-  const [ startTime, setStartTime ] = React.useState<any>(new Date());
-  const [ endDate, setEndDate ] = React.useState<any>(new Date());
-  const [ endTime, setEndTime ] = React.useState<any>(new Date());
-  const [input, setInput] = React.useState(["yes", "no"]);
+  const [account, setAccount] = useState('')
+  const web3 = new Web3Provider(window.ethereum)
+  const [title, setTitle] = useState('')
+  const [content, setContent] = useState('')
+  const [startDate, setStartDate] = useState(new Date())
+  const [startTime, setStartTime] = useState(new Date())
+  const [endDate, setEndDate] = useState(new Date())
+  const [endTime, setEndTime] = useState(new Date())
+  const [choices, setChoices] = useState(['yes', 'no'])
 
-  const inputRef = React.useRef();
-
-  const inputIsValid =
-    input.length >= 1 && input.every((field) => field.trim() !== "");
+  const inputRef = useRef()
 
   function handleChange(i, event) {
-    const values = [...input];
-    values[i] = event.target.value;
-    setInput(values);
+    const values = [...choices]
+    values[i] = event.target.value
+    setChoices(values)
   }
 
   function handleAdd() {
-    const values = [...input];
-    values.push("");
-    setInput(values);
+    const values = [...choices]
+    values.push('')
+    setChoices(values)
   }
 
   function handleRemove(i) {
-    const values = [...input];
-    values.splice(i, 1);
-    setInput(values);
+    const values = [...choices]
+    values.splice(i, 1)
+    setChoices(values)
   }
 
-  const submit = async() => {
-    try{
-      const receipt = await client.proposal(web3, account, {
+  const submit = async () => {
+    try {
+      // get current block of Gnosis network
+      const provider = await snapshot.utils.getProvider('100')
+      const block = await snapshot.utils.getBlockNumber(provider)
```

Comment:
Hard-coded the Gnosis network (100) since the poap strategy on the creative space uses it.

Reply:
That should work well. I've been seeing POAP testing mints on Polygon, so we'll stick a bookmark on this snippet.